auth/store.go

```diff
@@ -76,6 +76,9 @@ type AuthStore interface {
 	// UserGet gets the detailed information of a user
 	UserGet(r *pb.AuthUserGetRequest) (*pb.AuthUserGetResponse, error)
 
+	// UserRevoke revokes a role of a user
+	UserRevoke(r *pb.AuthUserRevokeRequest) (*pb.AuthUserRevokeResponse, error)
```

Comment:
I feel we should probable rename this to userRevokeRole (also userGrantRole) in another pr. userRevoke is easily confused with userDelete. also for roleRevoke -> roleRevokePermission. But it is fine for this pr.